### ISSUE TITLE
selector-class-pattern kebab-case 

### ISSUE THREAD

**Issue by h9lk** [stylelint/stylelint#5738]
### What steps are needed to reproduce the bug?

1. install any version after `stylelint@14.0.0 stylelint-config-standard@23.0.0`
2. in simplest cases you will get kebab-case like `.main-progress__bar `

### What Stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
    extends: [
        'stylelint-config-standard',
        'stylelint-config-rational-order',
    ],
};
```

### How did you run Stylelint?

CLI with `"lint:css": "stylelint **/*.sass"` in scripts

### Which version of Stylelint are you using?

14.1.0/14.0.1

### What did you expect to happen?

from my point when writing class as `.main-progress__cap` there should be no problems, but apparently everything is different in config, have to disable this rule in config, although in many cases it's useful

### What actually happened?

using class like this, which very strangely leads to permanent errors in selector-class-pattern, only in version `stylelint@13.13.1 stylelint-config-standard@22.0.0` no problems arise. Would continue to use this version, but Visual Studio Code swears that it's outdated and shows notifications about non-standard behavior - situation itself is very strange, you write `.main-progress_cap` same problem if `.main-progress-cap` not.

**Comment by ybiquitous** [comment]
@h9lk Thanks for using the template.

`stylelint-config-standard@23.0.0` has added the `selector-class-pattern` rule to check the kebab-case selector.

- https://github.com/stylelint/stylelint-config-standard/blob/480566337e15e4cd722bf63b5551ebdf55b1ba92/CHANGELOG.md#2300
- https://github.com/stylelint/stylelint-config-standard/blob/480566337e15e4cd722bf63b5551ebdf55b1ba92/index.js#L129-L134

Maybe, the following plugins can help you if you are using BEM:

- [stylelint-bem](https://github.com/merkle-open/stylelint-bem)
- [stylelint-selector-bem-pattern](https://github.com/simonsmith/stylelint-selector-bem-pattern)

**Comment by jeddy3** [comment]
@ h9lk Thanks for the report and for using the template.

> from my point when writing class as .main-progress__cap there should be no problems,

The config is designed to help people write CSS that looks like something you'd find in the examples of the specifications.

For more advanced use cases, e.g. an involved naming methodology like BEM, you'll need to extend and customise this config.

You should turn off the `selector-class-pattern` rule (using `null`) and I suggest using the [stylelint-selector-bem-pattern](https://www.npmjs.com/package/stylelint-selector-bem-pattern) plugin instead. It's very powerful and very good at enforcing BEM conventions.
